falcon/util/uri.py

```diff
@@ -345,8 +261,7 @@ def parse_query_string(query_string, keep_blank=False, csv=True):
     is_encoded = '+' in query_string or '%' in query_string
 
     # PERF(kgriffs): This was found to be faster than using a regex, for
-    # both short and long query strings. Tested on both CPython 2.7 and 3.4,
-    # and on PyPy 2.3.
+    # both short and long query strings. Tested on CPython 3.4.
```

Comment:
Maybe we should retest on CPython 3.5+, otherwise we would making claims based on a Python version that is no longer supported by this code.

Reply:
Agreed. However, I think that belongs in a separate patch if a regex ends up being faster on 3.5+. Do you want to benchmark that?

Reply:
Fine to do this in a separate PR. :+1: